src/sds_data_model/dataframe.py

```diff
@@ -4,18 +4,22 @@
 from inspect import ismethod, signature
 from logging import INFO, Formatter, StreamHandler, getLogger
 from pathlib import Path
-from typing import Any, Callable, Dict, List, Optional, Sequence, Type, TypeVar, Union
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Type, TypeVar, Union
 
 from pyspark.pandas import DataFrame as SparkPandasDataFrame
 from pyspark.pandas import Series as SparkPandasSeries
 from pyspark.pandas import read_excel
 from pyspark.sql import DataFrame as SparkDataFrame
 from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, explode, udf
+from pyspark.sql.types import ArrayType, FloatType, StringType
 from pyspark_vector_files import read_vector_files
 from pyspark_vector_files.gpkg import read_gpkg
 from typing_extensions import Self
+from shapely.wkt import loads
+from bng_indexer import calculate_bng_index, wkt_from_bng
```

Comment:
You aren't using `wkt_from_bng`

```suggestion
from bng_indexer import calculate_bng_index
```